docs/USAGE.md

```diff
@@ -12,19 +12,20 @@ how you may configure the settings.
 
 ## Emacs
 
- - Install [lsp-mode](https://github.com/emacs-lsp/lsp-mode)
- - Add the following to your `.emacs`:
+If you are using `use-package`, you can use something like this:
 
+``` emacs-lisp
+(use-package lsp-mode
+  :ensure t
+  :hook ((terraform-mode . lsp-deferred)))
 ```
-(lsp-register-client
- (make-lsp-client :new-connection (lsp-stdio-connection '("/path/to/terraform-ls/terraform-ls" "serve"))
-                  :major-modes '(terraform-mode)
-                  :server-id 'terraform-ls))
 
-(add-hook 'terraform-mode-hook #'lsp)
-```
+The above code installs `lsp-mode` and also enables it for the
+terraform-mode.
 
-The last line can instead be `(add-hook 'terraform-mode-hook #'lsp-deferred)` if you prefer the server to lazy load.
+The terraform client for lsp-mode supports various features like
+semantic tokens, code lens for references etc. There is more detailed
+documentation [here](https://emacs-lsp.github.io/lsp-mode/page/lsp-terraform-ls/).
```

Comment:
```suggestion
The `lsp-mode` language client for Terraform supports various features, such as
semantic tokens, code lens for references etc. There is more detailed
documentation [here](https://emacs-lsp.github.io/lsp-mode/page/lsp-terraform-ls/).
```
As far as I can tell emacs has a single language client implementation with different configurations, so I think this wording may better reflect that?

Reply:
There are two implementations available at Emacs side for language server:

- lsp-mode
- elgot

I believe the current wording is more clear, but I can change to the suggested one if you think that's more clear.

Reply:
> We can omit lsp-mod if the statement is true for both lsp-mode and elgot 

I'm not sure if the statement is true for `elgot`. From a quick look at the docs, it seems nobody has tested it for terraform-ls: https://github.com/joaotavora/eglot#connecting-to-a-server

That being said, even if it works - it will have very minimal features compared to `lsp-mode` as I made sure to support additional features for it - code lens, provides & module calls widget support etc specifically in `lsp-mode`.

>  but what I primarily meant above is that we shouldn't imply there's a "Terraform client" 

Ah, okay. I have made the suggested changes.